src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -5,91 +5,161 @@
-        [CLSCompliant(false)]
-        public uint GetUInt32() { throw null; }
-        [CLSCompliant(false)]
-        public ulong GetUInt64() { throw null; }
+        public JsonNumber() => _value = "0";
 
-        public bool TryGetByte(out byte value) { throw null; }
-        public bool TryGetInt32(out int value) { throw null; }
-        public bool TryGetInt16(out short value) { throw null; }
-        public bool TryGetInt64(out long value) { throw null; }
-        public bool TryGetSingle(out float value) { throw null; }
-        public bool TryGetDouble(out double value) { throw null; }
-        [CLSCompliant(false)]
-        public bool TryGetSByte(out sbyte value) { throw null; }
-        [CLSCompliant(false)]
-        public bool TryGetUInt16(out ushort value) { throw null; }
-        [CLSCompliant(false)]
-        public bool TryGetUInt32(out uint value) { throw null; }
-        [CLSCompliant(false)]
-        public bool TryGetUInt64(out ulong value) { throw null; }
+        public JsonNumber(string value) => SetFormattedValue(value);
+        
+        public JsonNumber(byte value) => SetByte(value);
+        
+        public JsonNumber(short value) => SetInt16(value);
+        
+        public JsonNumber(int value) => SetInt32(value);
+        
+        public JsonNumber(long value) => SetInt64(value);
+        
+        public JsonNumber(float value) => SetSingle(value);
+        
+        public JsonNumber(double value) => SetDouble(value);
+        
+        [CLSCompliant(false)]
+        public JsonNumber(sbyte value) => SetSByte(value);
+        
+        [CLSCompliant(false)]
+        public JsonNumber(ushort value) => SetUInt16(value);
+        
+        [CLSCompliant(false)]
+        public JsonNumber(uint value) => SetUInt32(value);
+        
+        [CLSCompliant(false)]
+        public JsonNumber(ulong value) => SetUInt64(value);
 
-        public void SetString(string value) { }
-        public void SetByte(byte value) { }
-        public void SetInt32(int value) { }
-        public void SetInt16(short value) { }
-        public void SetInt64(long value) { }
-        public void SetSingle(float value) { }
-        public void SetDouble(double value) { }
-        [CLSCompliant(false)]
-        public void SetSByte(sbyte value) { }
-        [CLSCompliant(false)]
-        public void SetUInt16(ushort value) { }
-        [CLSCompliant(false)]
-        public void SetUInt32(uint value) { }
-        [CLSCompliant(false)]
-        public void SetUInt64(ulong value) { }
+        public override string ToString() => _value; 
+        
+        public byte GetByte() => byte.Parse(_value);
+        
+        public short GetInt16() => short.Parse(_value);
+        
+        public int GetInt32() => int.Parse(_value);
+        
+        public long GetInt64() => long.Parse(_value);
+        
+        public float GetSingle() => float.Parse(_value);
+        
+        public double GetDouble() => double.Parse(_value);
```

Comment:
**Question:** Would escaped characters be supported? Especially since the `Utf8JsonWriter` escapes the `+` character by default (since it uses `JavascriptEncoder.Default`).

```C#
Console.WriteLine(double.Parse("25.1e+1")); // valid - 251
Console.WriteLine(double.Parse("25.1e\\u002B1")); // Fails
```